### PR TITLE
TNL-6510: Initialize elasticsearch for forums.

### PR DESCRIPTION
**Pull request by robrap**
Configuration Pull Request
---

Make sure that the following steps are done before merging

  - [x] A devops team member has commented with :+1:
  - [ ] are you adding any new default values that need to be overridden when this goes live?
    - [ ] Open a ticket (DEVOPS) to make sure that they have been added to secure vars.
    - [ ] Add an entry to the CHANGELOG.

### Sandbox
- [X] sandbox ([https://robrap.sandbox.edx.org/](https://robrap.sandbox.edx.org/))

### Post-review
- [x] Rebase and squash commits

### CODE REVIEW

**Comment by robrap** [comment]
@maxrothman: FYI: I'm provisioning a sandbox now to test this.

**Comment by robrap** [comment]
@maxrothman @dianakhuang: Discussions was properly initialized on the sandbox with no intervention.

**Review by dianakhuang — APPROVED** [review]
It might make sense to also get @maxrothman to test whether or not this is skipped in environments where it shouldn't be run.

**Review by maxrothman — COMMENTED** [review]
This looks good to me, though I'd like one more review from a member of @edx/devops.

**Review by maxrothman — APPROVED** [review]

**Comment by robrap** [comment]
@dianakhuang: Agreed.  But, because it is idempotent, it also really wouldn't matter at this point if it gets re-run.  I'll let @maxrothman determine how and if testing is necessary.

**Comment by maxrothman** [comment]
I'm reasonably confident that this'll be skipped when it should be, since `migrate_db` is only defined in playbooks where it should be run, so in all other cases, the `migrate_db is defined` check should fail. Still would like a second review though.

File: playbooks/roles/forum/tasks/deploy.yml [hunk 1]
```diff
@@ -60,6 +60,17 @@
     - install
     - install:app-requirements
 
+- name: initialize elasticsearch
+  command: bin/rake search:initialize
```
Comment (e0d) [inline playbooks/roles/forum/tasks/deploy.yml]: If the env script is correct, the complete relative path should not be necessary.  I guess my question is is this working around an issue with the env script?  If absolute safety is what you crave,  ```{{ forum_code_dir }}/bin/rake``` is probably better.

**Review by jibsheet — APPROVED** [review]

**Review by e0d — APPROVED** [review]
nit, but approve.

File: playbooks/roles/forum/tasks/deploy.yml (hunk 1 above)
Comment (robrap) [reply playbooks/roles/forum/tasks/deploy.yml]: @e0d: It was mainly copy/paste from [discovery code](https://github.com/edx/configuration/blob/master/playbooks/roles/discovery/tasks/main.yml#L97-L106).

Just tell me your preference and I'm happy to fix.

**Pull request merged**